Sponsors and SponsorContactInformation

catzkorn opened the pull request:
Created inital schema for `sponsors` and `sponsor_contact_information`. 

Used `ENUMS` to define sponsor_contact_information.role and sponsors.sponsorship_level.

Added some test data into the tables.

Created types to match schema representations. 

Implemented `ENUM` repesentation in types.go

Created a method that updates a based on SponsorContactInformation.ID

Added first tests that checks validation and update logic is working as intended.

Commit:
Create migration for sponsers and sponser_contact_information

Insert test data into db

Commit:
WIP sponsor and sponsor tiers

Commit:
Create UpdateSponsorContact method

Method allows a user to update all the information for a sponsor contact

Commit:
Update add_sponsors migrations

Commit:
Test updatesponsorcontact to check validation

Test also covers checking whether the method correctly updates a row.

Commit:
Add updatesponsorcontactinformation

Method updates a row by SponsorContactInformation.ID

Method validates it's inputs.

Commit:
Remove notes from Readme

bketelsen commented on conferences/migrations/3_add_sponsors.up.sql:
```diff
@@ -0,0 +1,160 @@
+  'platinum'
+);
+
+INSERT INTO sponsor (
```
Sponsors belong to conferences. Each year the list of sponsors may change, and the contact information for repeat sponsors may change.

bketelsen commented on conferences/types.go:
```diff
@@ -71,3 +71,56 @@ type Location struct {
+	Name             string
+	Address          string
+	Website          string
+	SponsorshipLevel SponsorshipLevel
```
Need the conference_id here for foreign key to conference table.

bketelsen reviewed (changes requested):
sponsorships need to have a relationship to a conference

Commit:
Add conference_id to sponsor

conference_id references conference.id

Commit:
Update Sponsor type to include ConferenceID

Remove ConferenceSlot slice from Conference

Add ConferenceID to ConferenceSlots

Commit:
Update SQL query to return conference_id

bketelsen reviewed (approved):
LGTM
